internal/openaiapi/handler_test.go

```
@@ -0,0 +1,326 @@
+	h := &Handler{
+		Provider:     &mockProvider{response: "Hello from the agent!"},
+		SystemPrompt: "You are a test assistant.",
+		AgentName:    "test-agent",
+	}
+
+	body := ChatCompletionRequest{
+		Model:    "docsclaw",
+		Messages: []ChatMessage{{Role: "user", Content: "Hi"}},
+	}
+	b, _ := json.Marshal(body)
+	req := httptest.NewRequest(http.MethodPost, "/v1/chat/completions", bytes.NewReader(b))
```

golangci-lint (2.12.1)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)